scripts/runnable_recipe_ray_vllm_weight_sync.py

```diff
@@ -491,270 +480,73 @@ def _setup_data(
-                tensor.cpu() if isinstance(tensor, torch.Tensor) else tensor
-                for tensor in postprocessed_results
-            ]
-
-            # Update circular queue and count full queue tries
-            # full_queue_data_discard = 0
-            while True:
-                try:
-                    self.rollout_queue.put_nowait(postprocessed_results)
-                    break
-                except QueueFull:
-                    self.rollout_queue.get()  # Remove the oldest item to make space
-                    # full_queue_data_discard += 1
-                    print(f"rollout queue full. Discarding data.")
-
-            if self._is_actor_zero:
-                # End timing the rollout step
-                time_total_rollout = time.perf_counter() - time_step_start
-
-                # TODO: training.get_memory_stats() crashes vLLM
-                # Log metrics
-                gpu_memory = {
-                    "allocated": torch.cuda.max_memory_allocated(device="cuda:0"),
-                    "reserved": torch.cuda.max_memory_reserved(device="cuda:0"),
-                    "active": torch.cuda.memory_stats(device="cuda:0").get(
-                        "active_bytes.all.peak", 0
-                    ),
-                }
-                time_total_rollout = time.perf_counter() - time_step_start
-                self._log_metrics(
-                    step_idx=idx,
-                    time_total_rollout=time_total_rollout,
-                    time_generate=time_generate,
-                    total_generated_tokens=total_generated_tokens,
-                    # full_queue_data_discard=full_queue_data_discard,
-                    gpu_memory=gpu_memory,
-                )
-
-
-class vLLMWorkerWrapper(Worker):
-    """vLLM Rollout Model worker for Ray."""
-
-    def __init__(self, *args, **kwargs):
-        import os
-
-        print(f"visible devices {os.environ['CUDA_VISIBLE_DEVICES']}")
-        print(f"device count {torch.cuda.device_count()}")
-        super().__init__(*args, **kwargs)
+    cfg = cfg
+    grpo_samples = cfg.grpo_samples
+    batch_size = cfg.batch_size
+    # _tokenizer = config.instantiate(cfg.tokenizer)
+    # collate_name = cfg.get(
+    #     "collate_fn", "torchtune.dev.grpo.data.padded_collate_rl"
+    # )
+    _tokenizer = AutoTokenizer.from_pretrained("Qwen/Qwen2.5-3B")
+    _tokenizer.pad_token = _tokenizer.eos_token
+    _tokenizer.padding_side = "right"
+    _dataloader = _setup_data(
+        cfg.dataset,
+        shuffle=cfg.shuffle,
+        batch_size=batch_size,
+        # collate_fn=collate_name,
+        rank=rank,
+        num_replicas=num_replicas,
+    )
 
-    def init_weight_update_group(self, master_address, master_port, rank, world_size):
-        from vllm.distributed.parallel_state import get_world_group
+    
+    from torchrl.envs import LLMEnv
+    llmenv = LLMEnv.from_dataloader(
+        dataloader=_dataloader,
+        tokenizer=_tokenizer,
```

Comment:
I have to use a hf tokenizer here as LLMEnv will call `batch_encode_plus` which the tune tokenizer doesn't have